htcanalyze/main.py

```diff
@@ -562,20 +568,25 @@ def print_results(htcanalyze: HTCAnalyze,
         else:
             results = htcanalyze.analyzed_summary(log_files)  # else
 
-    # Allow this to happen
+    # Allow to end program without output, if no scripts were analyzed
     if results is None:
         sys.exit(0)
 
-    work_with = results
     # convert result to list, if given as dict
     if isinstance(results, dict):
         work_with = [results]
+    else:
+        work_with = results
```

Comment:
/Please/ rename `work_with`. I don't know what it means, other than it's a copy of results. A better variable name and/or comment is needed to explain why we need a copy of `results` here rather than the original one.

Reply:
`work_with = [results] if isinstance(results, dict) else results`

Logically, it would be better to do:

`work_with = [results] if not isinstance(results, list) else results`

You want to convert it to a list if it isn't a list. Right now you are expecting that work_with is either a list or a dictionary, which probably makes sense. But logically speaking, you only want it to be a list if it isn't a list already.